### ISSUE TITLE
Unable to enable fingerprint verification of superuser

### ISSUE THREAD

**Issue by Skimige** [topjohnwu/Magisk#532]
[MagiskManager.log](https://github.com/topjohnwu/Magisk/files/2342481/MagiskManager.log)

Version : v17 Release & MagiskManager 5.9.0(138)

Got #529 then [fixed it](https://github.com/topjohnwu/Magisk/issues/529#issuecomment-417857031). Maybe this caused my issue.

Settings - "启用指纹验证"(fingerprint verification) - "验证失败"(Failed to authorize)

Don't know whether this log works, I'm sorry for that.

**Comment by topjohnwu** [comment]
There is something wrong with the fingerprint manager in your system, it reports authentication error and I really can't do much about that.

**Comment by PopeRigby** [comment]
I'm still having this problem. I keep removing my screen lock in my system settings and it doesn't fix the problem. How did you do it @hymnzzy?

**Comment by Miguel-Dorta** [comment]
I also have this problem. When I reset my biometric settings I get a message prompt from other apps when they try to use the fingerprint (or a fingerprint cypher), in my case was Keepass2Android. In this case they tell you to unlock the database with a password, disable and re-enable the fingerprint authentication.

That makes me think that the fingerprint authentication works by assigning something to each app. When you reset your biometric settings that thing assigned (maybe a token, a cypher, maybe having that app in a list, whatever) is not longer functional, and it throws a specific exception that, in the case of the Magisk Manager, it's not specifically caughs and just shows that error in a notification.

If that is the case, maybe the problem resides in [Magisk/app/src/main/java/com/topjohnwu/magisk/utils/FingerprintHelper.java](https://github.com/topjohnwu/Magisk/blob/master/app/src/main/java/com/topjohnwu/magisk/utils/FingerprintHelper.java). It would be great if you find a way to reset that token/cypher/whatever or implement a button in settings that let's you do that.

Remember, I'm not an Android programmer, I don't know how any of this works, I could be very wrong about that.

**Comment by hymnzzy** [comment]
@PopeRigby & @Miguel-Dorta .. What are the apps that are giving you this error?

**Comment by Miguel-Dorta** [comment]
Magisk Manager 7.2.0 (213)

Installed on:
- Samsung Galaxy S8 (SM-G950F, Exynos model)
- Android 9
- Kernel version: 4.4.111
- Android security patch level: 1 February 2019

I could previously use fingerprint verification, but I had to change my biometrics settings a while ago and I can't since that moment, so it worked fine.

![Screenshot showing the bug](https://i.imgur.com/Xidkplx.jpg)

**Comment by PopeRigby** [comment]
Magisk Manager 7.2.0 (213)

Installed on:
• Moto G5 Plus
• Android 8.1.0
• Kernel version: 3.18.71
• Android security patch level: October 1, 2018

**Comment by hymnzzy** [comment]
@PopeRigby & @Miguel-Dorta .. That's where my expertise ends.. All I can suggest is have you tried reinstalling the apps that are giving you errors? or at least clearing app cache?

**Comment by Miguel-Dorta** [comment]
@hymnzzy First, it doesn't work cleaning up the cache, second, that's a terrible "solution" (in quotes because it's not a solution)

**Comment by hymnzzy** [comment]
@Miguel-Dorta .. I hope you read my last comment which says "That's where my expertise ends" .. :)

**Comment by V10lator** [comment]
This issue is getting out of control. I just had the error while trying to authenticate an APP (from magisks superuser setting as fingerprint popup authentication didn't work) so I tried to disable fingerprint authentication in magisks settings but that gives me the same error and does not disable the option. I already tried serting device lock to none but magisk keeps being completely unusable. Any hint how to fix this?
